src/java/voldemort/client/protocol/admin/AdminClient.java

```diff
@@ -315,11 +319,17 @@ public void close() {
      */
     public boolean isClusterModified() {
         try {
-            Properties props = MetadataVersionStoreUtils.getProperties(metadataVersionSysStoreClient);
-            long retrievedVersion = MetadataVersionStoreUtils.getVersion(props,
-                                                                         SystemStoreConstants.CLUSTER_VERSION_KEY);
+            if (null == clusterVersion) {
+                // Then we have no choice to do a full cluster.xml equality check in order to know if the cluster is the same.
+                Cluster remoteCluster = getClusterFromBootstrapURL(clientConfig);
+                return currentCluster.equals(remoteCluster);
```

Comment:
Oh duh. Of course, you are totally right. And yes, I will also initialize the time before fetching the `remoteCluster`. Good call. Thanks a lot!